batch2/create-batch-tables.sql

```diff
@@ -94,6 +117,54 @@ CREATE INDEX batch_attributes_key_value ON `batch_attributes` (`key`, `value`(25
 
 DELIMITER $$
 
+CREATE TRIGGER attempts_before_update BEFORE UPDATE ON attempts
+FOR EACH ROW
+BEGIN
+  IF OLD.start_time IS NOT NULL AND (NEW.start_time IS NULL OR NEW.start_time > OLD.start_time) THEN
+    SET NEW.start_time = OLD.start_time;
+  END IF;
+
+  IF OLD.end_time IS NOT NULL AND (NEW.end_time IS NULL OR NEW.end_time > OLD.end_time) THEN
+    SET NEW.end_time = OLD.end_time;
+    SET NEW.reason = OLD.reason;
+  END IF;
+END $$
+
+CREATE TRIGGER attempts_after_update AFTER UPDATE ON attempts
+FOR EACH ROW
+BEGIN
+  DECLARE cores_mcpu INT;
+  DECLARE time_diff DOUBLE;
+  DECLARE cost_per_core_sec DOUBLE;
+  DECLARE new_msec_mcpu BIGINT;
+
+  SET cost_per_core_sec = 0.01 / 3600;
```

Comment:
As mentioned above, this doesn't account for the cost of the disk.

Reply:
SSD PD is $0.17/GB-month.  So for 8 cores, 100G, the cost of disk per core hour is:

```
0.170 / 30 / 24 * 12.5
.00295138888888888887
```

So the total instance cost per core-hour is 0.0013.

Reply:
Shouldn't there be a factor of 100 in there?

0.17 GB/month * 100 GB * (1 month / 30 days) * (1 day / 24 hours) * (1 machine / 8 cores)

Ok. So that's where the 12.5 comes from then.

Reply:
This should be 0.013 (0.01 + 0.003)

Reply:
Yeah, the 12.5 is 100 / 8.